examples/fidelity.py

```diff
@@ -0,0 +1,86 @@
+"""Simulates the fidelity estimation.
+
+Direct Fidelity Estimation from Few Pauli Measurements
+https://arxiv.org/abs/1104.4695
+
+Practical characterization of quantum devices without tomography
+https://arxiv.org/abs/1104.3835
+"""
+
+import cirq
+import heapq
+import itertools
+import math
+import numpy
+
+
+def build_circuit():
+  # Builds an arbitrary circuit to test. The circuit is non Clifford to show the
+  # use of simulators.
+  qubits = cirq.LineQubit.range(3)
+  circuit = cirq.Circuit(
+      cirq.Z(qubits[0])**0.25,  # T-Gate, non Clifford.
+      cirq.X(qubits[1])**0.123,
+      cirq.X(qubits[2])**0.456)
+  return circuit, qubits
+
+
+def simulate_trace(circuit, pauli_gates, qubits, noise):
+  simulator = cirq.DensityMatrixSimulator(noise=noise)
+
+  n = len(pauli_gates)
+  d = 2**n
+
+  trace = 0
+  for x in range(d):
+    xbin = numpy.binary_repr(x, width=n)
+
+    pauli_string = cirq.PauliString(dict(zip(qubits, pauli_gates)))
+    display = cirq.approx_pauli_string_expectation(pauli_string, num_samples=n)
```

Comment:
why is num_samples set to the number of qubits?

Reply:
I had misread first paragraph of second column of second page of the paper: https://arxiv.org/pdf/1104.3835.pdf

The repetition is *not* within a rotation, so set to 1.